### ISSUE TITLE
build_taxonomies throws fatal error (Unsupported operand types)

### ISSUE THREAD

**Issue by FreekGille** [Auctollo/google-sitemap-generator#62]
The function `build_taxonomies` takes two arguments being `$gsg` and `$taxonomy`. The function then proceeds to do the following:

```php
$offset = $taxonomy;
$links_per_page = $gsg->get_option( 'links_page' );
if ( strpos( $taxonomy, '-' ) !== false ) {
    $offset = substr( $taxonomy, strrpos( $taxonomy, '-' ) + 1 );
    $taxonomy = str_replace( '-' . $offset, '', $taxonomy );
}
$offset = ( --$offset ) * $links_per_page;
```

I use a custom post category which can be `cocktails` or `creators`.
I'm not sure why the check for a `-` is in these but for me the last line throws an error: Unsupported operand types: int * string.

**Comment by FreekGille** [comment]
So I've just done a bit more digging. The error is triggered when PetalBot (some Huawei crawler) checks sitemap URLs without following the structure of the main sitemap.xml. An example is them searching for `/sitemap-tax-creators.xml` while the correct format would be `/sitemap-tax-creators-1.xml`. 
In the faulty case, `$offset` will remain `creators` (in this case) and the valid case, the if-statement will trigger and split the taxonomy name and offset integer. 

I believe an else statement ought to be added to the if-statement that sets `$offset` to 1. Happy to add a PR if wanted,

**Comment by FreekGille** [comment]
@saurabhd Any opinion on this?

**Comment by saurabhd** [comment]
Hi @FreekGille this is fixed in our new 4.1.7 version, please check. Thanks for showing interest and providing all the information in depth manner.

**Comment by FreekGille** [comment]
Great! Thanks for fixing this issue! I'll close this issue.
